python/paddle/distributed/fleet/layers/mpu/random.py

```diff
@@ -51,10 +51,11 @@ def add(self, name, seed):
         self.seeds_.add(seed)
         if name in self.states_:
             raise ValueError(f'state {name} already exists')
-        orig_rng_state = paddle.get_rng_state()
+        orig_rng_state_index = paddle.incubate.get_rng_state(use_index=True)
+        # register a new state and set that state with the seed, store the indices into states_
+        self.states_[name] = paddle.incubate.register_rng_state_as_index()
         paddle.seed(seed)
```

Comment:
Do we reorder `paddle.seed()` and `self.states_[name] = ...` in the wrong order here?

Reply:
No, the order of operations is correct as implemented. Here's a breakdown of the process:

Original Code:
```
orig_rng_state = paddle.get_rng_state()
paddle.seed(seed)
self.states_[name] = paddle.get_rng_state()
paddle.set_rng_state(orig_rng_state)
```
In this snippet, we first save the original RNG state. Then, we set a new seed and update `self.states_[name]` with the new RNG state. Finally, we restore the original RNG state.


Revised Code:
```
orig_rng_state_index = paddle.incubate.get_rng_state(use_index=True)
# register a new state and set that state with the seed, store the indices into states_
self.states_[name] = paddle.incubate.register_rng_state_as_index()
paddle.seed(seed)
paddle.incubate.set_rng_state(orig_rng_state_index, use_index=True)
```
In the revised code, we first obtain the index of the original RNG state. We then register a new RNG state and update `self.states_[name]` with the index of this new state. Then, we set the seed for the new state. Finally, we switch back to the old state.